### DISCUSSION TITLE
Syntax highlighting spans where the end condition depends on the begin condition

### DISCUSSION THREAD

**Discussion by khhelland** [icsharpcode/AvalonEdit#330] [Q&A] [answered]
Hi, I'm using AvalonEdit for a Lua editor. I want to properly highlight multiline comments, but the end condition can depend on the begining condition.
The format is 
"
--[[
 Comment
]]
"
, but also
"
--[===[
Comment
]===]
"
where the number of "="s must match.

I've tried using a span like so:
```
	<Span color="Comment" multiline="true" begin="--\[(=*)\[" end="\]$1\]"/\>
```
but this does not work. 

Is it possible to use a group captured in the "begin" in the "end"?
And if not is there another way to accomplish this?

**Comment by siegfriedpammer** [comment, marked as answer]
The grammar you are working with is too complex for regular expressions. You will have to implement your own lexer-based highlighting based on `IHighlighter`.

See https://github.com/icsharpcode/SharpDevelop/blob/master/src/AddIns/BackendBindings/CSharpBinding/Project/Src/CSharpSemanticHighlighter.cs for an example of a semantic highlighter for C#.

Hope this helps!
